packages/@aws-cdk/aws-stepfunctions-tasks/test/send-to-queue.test.ts

```diff
@@ -25,4 +30,74 @@ test('publish to queue', () => {
       'MessageDeduplicationId.$': '$.deduping'
     },
   });
+});
+
+test('message body can come from state', () => {
+  // WHEN
+  const pub = new sfn.Task(stack, 'Send', {
+    task: new tasks.SendToQueue(queue, {
+      message: sfn.DataField.fromStringAt('$.theMessage')
+    })
+  });
+
+  // THEN
+  expect(stack.node.resolve(pub.toStateJson())).toEqual({
+    Type: 'Task',
+    Resource: 'arn:aws:states:::sqs:sendMessage',
+    End: true,
+    Parameters: {
+      'QueueUrl': { Ref: 'Queue4A7E3555' },
+      'MessageBody.$': '$.theMessage',
+    },
+  });
+});
+
+test('message body can be an object', () => {
+  // WHEN
+  const pub = new sfn.Task(stack, 'Send', {
+    task: new tasks.SendToQueue(queue, {
+      messageObject: {
+        literal: 'literal',
+        SomeInput: sfn.DataField.fromStringAt('$.theMessage')
+      }
+    })
+  });
+
+  // THEN
+  expect(stack.node.resolve(pub.toStateJson())).toEqual({
+    Type: 'Task',
+    Resource: 'arn:aws:states:::sqs:sendMessage',
+    End: true,
+    Parameters: {
+      QueueUrl: { Ref: 'Queue4A7E3555' },
+      MessageBody: {
+        'literal': 'literal',
+        'SomeInput.$': '$.theMessage',
+      }
+    },
+  });
+});
+
+test('message body object can contain references', () => {
```

Comment:
Would be nice to verify that fields can also be embedded within strings via simple concat: `Hello, ${sfn.DataField.fromStringAt('$.World')}!`

Reply:
In fact that is not allowed, but I will have to make sure that's true then.